core/src/main/java/io/micronaut/core/optim/AotHelper.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.optim;
+
+import io.micronaut.core.annotation.Internal;
+
+/**
+ * AOT-specific helper.
+ *
+ * @since 4.2.0
+ * @author Jonas Konrad
+ */
+@Internal
+public final class AotHelper {
```

Comment:
move it to the same package as SoftServiceLoader and make package private

Reply:
my intention was to add more similar flags and collect them here but maybe that's overkill. ive moved the class.